ideas/276-fast-discovery.md

```diff
@@ -0,0 +1,142 @@
+---
+id: 276-fast-discovery
+title: Fast discovery
+status: draft
+created: 2018-06-22
+category: core
+lead-contributor: dshulyak
+contributors:
+        - dshulyak
+exit-criteria: yes
+success-metrics: yes
+clear-roles: yes
+future-iteration: ?
+roles-needed:
+    - Go Developer
+okrs:
+    - "[PO]: Objective: Messaging is reliable"
+    - "[P2]: KR: Cluster can handle 500 concurrent users"
+---
+
+## Preamble
+
+    Idea: #276-fast-discovery
+    Title: Fast discovery
+    Status: draft
+    Created: 2018-06-22
+
+## Summary
+
+Previously we added support for discovery v5 in status client.
+While it allows us to discover peers in under 5 seconds, and consumes on average 50kb of traffic (sum of inbound and outound).
+This picture completely changes with more concurrent clients. If 100 clients joins simultaneously discovery becomes
+very slow and not reliable. For instance it can consume as much as 2mb in 10s while not finding required number of peers.
+
+## Swarm Participants
+
+- Lead Contributor: dshulyak
+
+## Product Overview
+
+We will implement protocol that takes into consideration problems mentioned in summary of this document.
```

Comment:
I would add more information how it's gonna play with the current Discovery V5. Do we want it to replace Discover V5? Will it run along Discovery V5 on the server nodes but it will be the only discovery protocol on mobile nodes? 

For example, if we want to add support for LES, we need to have Ethereum discovery protocol running so it's worth to consider such a case as well.

Reply:
updated with several alternatives. the only realistic thing however would be to implement proxy